formspree/forms/models.py

```diff
@@ -42,6 +60,16 @@ def __init__(self, email, host):
     def __repr__(self):
         return '<Form %s, email=%s, host=%s>' % (self.id, self.email, self.host)
 
+    def get_random_like_string(self):
+        if not self.id:
+            raise Exception("this form doesn't have an id yet, commit it first.")
+        return int2bigstring(self.id)
```

Comment:
Is this better than just creating a random string for each form, and saving it in a separate column? Then here the result would just be self.random_str, and get_form_by_random_string() would just query for a matching random_str. 

In the scenario above, self.id is likely to be non-random, and the int2bigstring is not really an encryption function, and the algorithm is published here. So I could imagine an exploit where the attacker, wanting to send emails to one of our users, first POSTs to formspree.io/targetemail, and then guesses the form.id, calculates the random_like_string, and sends a GET request to the corresponding confirm endpoint. Basically a user could confirm and then send emails to an email address without owning that email.